### PR TITLE
translation: Update the figures for the chapter of array and linked list

### PR DESCRIPTION
**Pull request by krahets**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [x] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [x] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

### CODE REVIEW

File: docs-en/chapter_array_and_linkedlist/array.assets/array_memory_location_calculation.png
Comment (K3v123) [inline docs-en/chapter_array_and_linkedlist/array.assets/array_memory_location_calculation.png:1]: I think its better to stick with one, either "memory address" or "memory location" to make it more consistent.

File: docs-en/chapter_array_and_linkedlist/array.assets/array_remove_element.png
Comment (K3v123) [inline docs-en/chapter_array_and_linkedlist/array.assets/array_remove_element.png:1]: optional,
making the last element 4 invalid

->
rendering the last element 4 meaningless.

File: docs-en/chapter_array_and_linkedlist/array.assets/array_memory_location_calculation.png
Comment (krahets) [reply docs-en/chapter_array_and_linkedlist/array.assets/array_memory_location_calculation.png:1]: They are often used interchangeably. 

- "memory address" emphasizes the access method—how the system identifies and reaches a particular spot in memory.
- "memory location" might be more about the content or significance of what is stored at that address. 

In most contexts, though, the distinction is minor, and the terms are used to mean essentially the same thing: where data resides in memory.

File: docs-en/chapter_array_and_linkedlist/array.assets/array_remove_element.png
Comment (krahets) [reply docs-en/chapter_array_and_linkedlist/array.assets/array_remove_element.png:1]: I think invalid is more precise here. "Rendering" is a great alternative!

**Review by SamJin98 — APPROVED** [review]
Looks good to me.

**Review by K3v123 — APPROVED** [review]

**Pull request merged**